tests/unit/changeset-test.js

```diff
@@ -1057,6 +1057,60 @@ module('Unit | Utility | changeset', function(hooks) {
     assert.deepEqual(get(dummyModel, 'org'), expectedResult.org, 'should set value');
   });
 
+
+  /**
+   * #pendingChanges
+   */
+
+  test('#pendingChanges up to date if changeset is invalid', async function(assert) {
+    let dummyChangeset = Changeset(dummyModel, dummyValidator);
+    dummyChangeset.set('name', 'a');
+
+    assert.equal(get(dummyModel, 'name'), undefined, 'precondition');
+    assert.equal(get(dummyChangeset, 'isValid'), false, 'can be invalid');
+
+    let pendingChanges = dummyChangeset.pendingData;
+
+    assert.equal(get(pendingChanges, 'name'), 'a', 'should apply changes');
+  });
+
+  test('#pendingChanges up to date if changeset is valid', async function(assert) {
+    let dummyChangeset = Changeset(dummyModel);
+    dummyChangeset.set('name', 'foo');
+
+    assert.equal(get(dummyModel, 'name'), undefined, 'precondition');
+    assert.equal(get(dummyChangeset, 'isValid'), true, 'can be valid');
+
+    let pendingChanges = dummyChangeset.pendingData;
+
+    assert.equal(get(pendingChanges, 'name'), 'foo', 'should apply changes');
+  });
+
+  test('#pendingChanges up to date after saved changeset modified again', async function(assert) {
+    let dummyChangeset = Changeset(dummyModel);
+    dummyChangeset.set('name', 'foo');
+
+    assert.equal(get(dummyModel, 'name'), undefined, 'precondition');
+    assert.equal(get(dummyChangeset, 'isValid'), true, 'can be valid');
+
+    assert.equal(get(dummyModel, 'name'), undefined, 'original data is not applied');
```

Comment:
Can we move this assertion after the `pendingData` check?

Reply:
👍